Announcing the removal of the custom Logstash distribution (#1576)
opensearch-project/project-website: _posts/2023-05-11-Discontinuing-custom-Logstash-distribution.md

---
layout: post
title:  "Discontinuing the custom Logstash distribution"
authors:
- dlv
date: 2023-05-11 12:30:00 -0500
---

> One of the core [OpenSearch project principles](https://opensearch.org/about.html) is to be used everywhere. 

Author: I used the text "used everywhere" because that is exactly what I see in the project principals page: https://opensearch.org/about.html.

Perhaps this?

> One of the core [OpenSearch project principles](https://opensearch.org/about.html) includes being used everywhere: in businesses, software, and projects.

> One of the core [OpenSearch project principles](https://opensearch.org/about.html) is to be used everywhere. 

Reviewer: One of the core OpenSearch Project principles is that OpenSearch can be used everywhere: in businesses, software, and projects.

> meta_description: OpenSearch project will continue to provide Logstash plugins while discontinuing the custom distribution.

Reviewer: ```suggestion
meta_description: The OpenSearch Project will continue to provide Logstash plugins while discontinuing the custom distribution.
```

> One of the core [OpenSearch project principles](https://opensearch.org/about.html) is to be used everywhere.

Reviewer: ```suggestion
One of the core [OpenSearch Project principles](https://opensearch.org/about.html) is to be used everywhere.
```

> One of the core [OpenSearch project principles](https://opensearch.org/about.html) is to be used everywhere.

Reviewer: This sentence reads as though it's missing information. What core principle?

To help OpenSearch be used everywhere, we provide two Logstash plugins to allow OpenSearch users to ingest data to and from OpenSearch using Logstash: [logstash-output-plugin](https://github.com/opensearch-project/logstash-output-opensearch) and [logstash-input-plugin](https://github.com/opensearch-project/logstash-input-opensearch). 

> We have also provided a custom distribution of Logstash with these plugins embedded. 

Author: I think "provided" is slightly more accurate. We don't develop the distribution. We run a release of it.

An alternative to "provided" might be "release."

> We also release a custom distribution of Logstash with these plugins embedded.

We believe that OpenSearch users will continue to benefit from our plugins using only the Apache-licensed Logstash distribution from Elastic and plan to discontinue the custom Logstash distribution by October 2023.

## Reasons to discontinue

Many users are downloading the custom OpenSearch distribution of Logstash. 

> So it does provide some convenience and value to OpenSearch users. 

Reviewer 2: ```suggestion
"... and we acknowledge that it does provide some convenience and value to OpenSearch users."
```
I think you can run the first two sentences together with a comma

However, we find some compelling reasons to discontinue the distribution.

> One of the reasons is to help users get updates quickly. 

Author: It's not the plugins that getting updates out. It is getting out updates to the users of our plugins.

I'll try this:

> One of the reasons is to help users of these plugins get updates quickly.

> Many users are downloading the custom OpenSearch distribution of Logstash, and we acknowledge that it does provide some convenience and value to OpenSearch users. 

Reviewer: Instead of "are downloading", either "download" or "have downloaded".

> Because the OpenSearch distribution is based on the Apache-licensed distribution provided by Elastic, users cannot get the updates immediately after they are released. 

Reviewer 2: This sentence confuses me. On line 13 we tell readers that our two plugins "use the Apache-licensed Logstash distribution from Elastic." But here it says the [custom?] distribution is based on the same "Apache-licensed distribution provided by Elastic." What exactly is the key difference between the plugins and the custom distribution that gives the plugins quicker updates?
Also, is it OK to start this sentence: "Because the custom OpenSearch distribution is based on ..." Having the word "custom" helps me distinguish between the distribution and the plugins.

Author: I'm not quite sure the best way to put this either. I pushed a change with another attempt.

I reworded to:

> Because the custom distribution is based on the Apache-licensed distribution provided by Elastic, users cannot get the updates immediately after Elastic releases them. 
Instead, they have to wait for the OpenSearch project to repackage the updates.

Thoughts?

> They have to wait for the OpenSearch project to repackage the updates. 

Reviewer 2: ```suggestion
Instead, they have to wait for the OpenSearch project to repackage the updates. 
```

> This is especially important for security patches which the OpenSearch project aims to provide as soon as possible.

Reviewer 2: ```suggestion
This is especially important for security patches, when getting these updates as soon as possible is a consideration.
```

> Another reason is to provide clarity for users on who is supporting these tools. 

Reviewer 2: ```suggestion
Another reason is that, by discontinuing the distribution, it becomes perfectly clear to users who is responsible for supporting the tools they use. 
```

Author: I really like this suggestion. Thanks!

> With this change, the direction to users for support is clear. 

Reviewer 2: Line 26 makes line 27 unnecessary.

> Instead, they have to wait for the OpenSearch project to repackage the updates.

Reviewer: ```suggestion
Instead, they have to wait for the OpenSearch Project to repackage the updates.
```

> If a user needs support for Logstash features or bundled Logstash plugins, they go to the main [Logstash support](https://discuss.elastic.co/c/elastic-stack/logstash/14). 

Reviewer 2: ```suggestion
When a user needs support for Logstash features or bundled Logstash plugins, they would typically go to the [Logstash discuss](https://discuss.elastic.co/c/elastic-stack/logstash/14) page. 
```

> When they have an issue with the OpenSearch plugins they know to seek support from the OpenSearch project.

Reviewer 2: ```suggestion
When they have an issue with the OpenSearch plugins, however, they know to seek support from the OpenSearch project.
```

Finally, the main Logstash distribution already supports downloads for a variety of platforms. 
For example, it supports Linux, macOS, and RPM installations. 

> The current OpenSearch distribution lacks many of these distributions.

Reviewer 2: ```suggestion
The custom OpenSearch distribution, on the other hand, lacks many of these distributions.
```

> When they have an issue with the OpenSearch plugins, however, they know to seek support from the OpenSearch project.

Reviewer: ```suggestion
When they have an issue with the OpenSearch plugins, however, they know to seek support from the OpenSearch Project.
```

Installing the OpenSearch plugins in the main Logstash distribution is straightforward. 

> Because it is quite easy, we believe that the reasons to discontinue outweigh the convenience of having the distribution. 

Reviewer 2: ```suggestion
And given the ease of installation, along with the reasons we've already listed, we believe that a decision to discontinue the distribution outweighs any degree of convenience associated with it. 
```

> Discontinuing this distribution will let us focus on deliver great features in our core products and the Logstash plugins.

Reviewer 2: ```suggestion
Discontinuing this distribution will also free up time and resources so we can better focus on delivering great features in our core products and the Logstash plugins.
```

## Going forward

The OpenSearch project will continue to maintain the existing Logstash plugins for OpenSearch. 

> Users will be able to [download Logstash](https://www.elastic.co/downloads/logstash) from Elastic. 

Reviewer 2: ```suggestion
To install the plugins, first [download Logstash](https://www.elastic.co/downloads/logstash) from Elastic and then run the following commands.
```

> And given the ease of installation, along with the reasons we've already listed, we believe that a decision to discontinue the distribution outweighs the convenience associated with it.  

Reviewer: ```suggestion
And given the ease of installation, along with the reasons we've already mentioned, we believe that the benefits of discontinuing the distribution outweigh its associated convenience.  
```

> And given the ease of installation, along with the reasons we've already listed, we believe that a decision to discontinue the distribution outweighs the convenience associated with it.  

Reviewer: Confirm that my changes retain the intended meaning.

> Then run the following commands to install the plugins.

Reviewer 2: You can delete this since it's moved above to line 42.

> The OpenSearch project will continue to maintain the existing Logstash plugins for OpenSearch.

Reviewer: ```suggestion
The OpenSearch Project will continue to maintain the existing Logstash plugins for OpenSearch.
```

> For the logstash-output-opensearch plugin, run:

Reviewer: Above, this is "logstash-output-plugin". Is this something different?

Author: I will correct this above. It should be `logstash-output-plugin`.

```
bin/logstash-plugin install logstash-output-opensearch
```

> And to install the logstash-input-opensearch plugin, run:

Reviewer: Same. Above, this is "logstash-input-plugin".

Author: I will correct this above. It should be `logstash-input-plugin`.

```
bin/logstash-plugin install logstash-input-opensearch
```

> In the coming weeks, we will improve the documentation for using these plugins with the Logstash distribution and start to remove the download links for the existing distribution. 

Reviewer: ```suggestion
In the coming weeks, we will improve the documentation on using these plugins with the Logstash distribution and start to remove the download links for the existing distribution. 
```

> In the coming weeks, we will improve the documentation for using these plugins with the Logstash distribution and start to remove the download links for the existing distribution. 

Reviewer: ```suggestion
In the coming weeks, we will improve the documentation on using these plugins with the Logstash distribution and start to remove the download links for the existing distribution. 
```

We plan to stop releasing new versions by the end of July 2023. 

> And we will retain existing download URLs through October 2023 to give time for users to update any automation they have in place.

Reviewer: ```suggestion
And we will retain existing download URLs through October 2023 to give you time to update any automation you have in place.
```

> And we will retain existing download URLs through October 2023 to give time for users to update any automation they have in place.

Reviewer: This could use a concluding sentence. What do we want the reader to do next or see for more information?

meta_description: The OpenSearch Project will continue to provide Logstash plugins while discontinuing the custom distribution.
---

> One of the core [OpenSearch Project principles](https://opensearch.org/about.html) includes being used everywhere: in businesses, software, and projects.

Reviewer: "One of the core OpenSearch Project principles is that OpenSearch can be used everywhere: in businesses, software, and projects."
